lib/elixir/lib/file.ex

```diff
@@ -1335,4 +1335,9 @@ defmodule File do
 
   defp open_defaults([], true),  do: [:binary]
   defp open_defaults([], false), do: []
+
+  defp maybe_to_string(path) when is_binary(path),
```

Comment:
I would rather do the opposite. Always convert unless you have a pid. Otherwise this code is wrong, a char list is a valid argument and now it no longer works. :)